src/mscorlib/shared/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -2230,7 +2338,7 @@ private unsafe void WriteEventString(EventLevel level, long keywords, string msg
                         data.Ptr = (ulong)msgStringPtr;
                         data.Size = (uint)(2 * (msgString.Length + 1));
                         data.Reserved = 0;
-                        m_provider.WriteEvent(ref descr, null, null, 1, (IntPtr)((void*)&data));
+                        m_provider.WriteEvent(ref descr, IntPtr.Zero, null, null, 1, (IntPtr)((void*)&data));
```

Comment:
I believe this will causes crashes (In native code in WriteEvent() we expect EventHandles to be well defined).   This code is used when EventSource logs error messages about the EventSource itself.  

The easiest way to defined the Event with ID zero to be this event (it is already sort of used this way.  

      m_eventData[0].Handle for this.  

Note that this 'works' OK (it will no longer crash), but we should also fix CreateManifestAndDescriptors to provide a better name, that is change 
    eventData[0].Name = ""

to 
     eventData[0].Name = "EventSourceMessage"

and provide it with meta-data (you can do this by setting eventData[0].Parameters to a method that has a one string argument with name 'message' and your code will generate the right meta-data for it.